### PR TITLE
Allow listBuckets request to be signed with region configured in config.json

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

Allow listBuckets request to be signed with region configured in config.json

## Motivation and Context

https://github.com/minio/minio/issues/3373

## How Has This Been Tested?

manual

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/bucket-handlers.go [hunk 1]
```diff
@@ -171,8 +171,11 @@ func (api objectAPIHandlers) ListBucketsHandler(w http.ResponseWriter, r *http.R
 
 	// ListBuckets does not have any bucket action.
 	if s3Error := checkRequestAuthType(r, "", "", "us-east-1"); s3Error != ErrNone {
-		writeErrorResponse(w, r, s3Error, r.URL.Path)
-		return
+		// Clients like boto3 send listBuckets() call signed with region that is configured.
+		if s3Error := checkRequestAuthType(r, "", "", serverConfig.GetRegion()); s3Error != ErrNone {
```
Comment (harshavardhana) [inline cmd/bucket-handlers.go]: This falblack should be only done if s3Error == ErrInvalidRegion

**Comment by harshavardhana** [comment]
Looks like one more unrelated patch is brought in here.. can you remove @krishnasrinivas ?

File: cmd/bucket-handlers.go [hunk 2]
```diff
@@ -171,10 +171,17 @@ func (api objectAPIHandlers) ListBucketsHandler(w http.ResponseWriter, r *http.R
 
 	// ListBuckets does not have any bucket action.
 	if s3Error := checkRequestAuthType(r, "", "", "us-east-1"); s3Error != ErrNone {
+	}
```
Comment (harshavardhana) [inline cmd/bucket-handlers.go]: looks like something happened.. here.

Comment (krishnasrinivas) [reply cmd/bucket-handlers.go]: oops, fixed

**Pull request merged**